tests/loan_protocol.rs

```diff
@@ -610,217 +528,34 @@ async fn lend_and_dynamic_liquidate() {
             .await
             .unwrap();
 
-        client
-            .send_raw_transaction(&liquidation_transaction)
-            .await
-            .expect("could not liquidate with valid proof of dip");
+        verify_collateral_output_spend(&liquidation_transaction, &loan_transaction)
+            .expect("dynamic liquidation transaction cannot spend from collateral contract");
     }
 }
 
-fn init_logger() {
-    // force enabling log output
-    let _ = env_logger::builder().is_test(true).try_init();
-}
-
-async fn blinding_key(client: &elements_rpc::Client, address: &Address) -> Result<SecretKey> {
-    let master_blinding_key = client.dumpmasterblindingkey().await?;
-    let master_blinding_key = hex::decode(master_blinding_key)?;
-
-    let sk = derive_blinding_key(master_blinding_key, address.script_pubkey())?;
-
-    Ok(sk)
-}
-
-async fn find_inputs(
-    client: &elements_rpc::Client,
-    asset: AssetId,
-    amount: Amount,
-) -> Result<Vec<Input>> {
-    let inputs = client.select_inputs_for(asset, amount, false).await?;
-
-    let master_blinding_key = client.dumpmasterblindingkey().await?;
-    let master_blinding_key = hex::decode(master_blinding_key)?;
-
-    let inputs = inputs
-        .into_iter()
-        .map(|(txin, tx_out)| {
-            let input_blinding_sk =
-                derive_blinding_key(master_blinding_key.clone(), tx_out.script_pubkey.clone())?;
-
-            Result::<_, anyhow::Error>::Ok(Input {
-                txin,
-                original_txout: tx_out,
-                blinding_key: input_blinding_sk,
-            })
+fn verify_collateral_output_spend(
```

Comment:
This is actually more generic than what you are saying in the function name.

This verifies that one transaction spends from the other.

I think you may want to be more specific here no? We actually want to verify that we are spending a very specific contract and not just any output. This will return true if `spend_transaction` spends a change output from `loan_transaction`.